src/glmtools.jl

```diff
@@ -58,11 +87,34 @@ glmvar(::Gamma, ::Link, μ, η) = abs2(μ)
 glmvar(::Normal, ::Link, μ, η) = 1
 glmvar(::Poisson, ::Link, μ, η) = μ
 
+"""
+    mustart(d, y, wt)
+
+Derive starting values for the mu vector.
+
+#### Arguments:                                                                             
+* `d`: Distribution, can be any one of `::Binomial`, `::Gamma`, `::Normal` or `::Poisson`  * `y`:                                                                                     * `wt`:
+"""
+function mustart end
+    
 mustart(::Binomial, y, wt) = (wt*y + oftype(y,0.5))/(wt + one(y))
 mustart(::Gamma, y, wt) = y
 mustart(::Normal, y, wt) = y
 mustart(::Poisson, y, wt) = y + oftype(y, 0.1)
 
+"""
+    devresid(d, y, μ, wt)
+
+Return a vector of squared deviance residuals.
+
+#### Arguments:
+* `d`: Distribution, can be any one of `::Binomial`, `::Gamma`, `::Normal` or `::Poisson`
+* `y`: 
+* `μ`:
+* `wt`:
+"""
```

Comment:
y is the response vector
mu is the predicted mean vector
wt is the weight vector